### DISCUSSION TITLE
如果前端用React写，会渲染很多很多的资料夹跟页面.....

### DISCUSSION THREAD

**Discussion by punkcanyang** [IntensiveCoLearning/Web3-URL#140] [General]
如果前端用React写，会渲染很多很多的资料夹跟页面.....
这种状况下，要如何放上EVM？

**Comment by DerickIT** [comment]
同问，所以想看看他们那个带有样式和执行逻辑的官网是如何实现的？
1.可能后续会提供缓存主流UI库的调用链接？类似于引入主流cdn提供的UI库
2，关于代码执行逻辑的简化，可能会实现与solidity紧密结合的类似htmx那种后端渲染好，前端就显示结果的功能？  webassembly
还有数据库，感觉要便于开发，并且能承载大量内容，这个还有很多事情要做啊

**Reply by qzhodl** [reply]
官网的repo: https://github.com/ethstorage/web3url-website/

**Comment by qzhodl** [comment]
前端的网页的渲染可以都交给js来完成，js也存储在链上。只是之前js请求数据需要通过服务器，现在通过智能合约请求数据。
